lib/DxilDia/DxcPixDxilDebugInfo.cpp

```diff
@@ -0,0 +1,115 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxcPixDxilDebugInfo.cpp                                                   //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Defines the main class for dxcompiler's API for PIX support.              //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxc/Support/WinIncludes.h"
+
+#include "dxc/Support/exception.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/DebugInfo.h"
+#include "llvm/IR/DebugInfoMetadata.h"
+
+#include "DxcPixLiveVariables.h"
+#include "DxcPixDxilDebugInfo.h"
+
+STDMETHODIMP dxil_debug_info::DxcPixDxilDebugInfo::GetLiveVariablesAt(
+  _In_ DWORD InstructionOffset,
+  _COM_Outptr_ IDxcPixDxilLiveVariables **ppLiveVariables)
+{
+  return m_LiveVars->GetLiveVariablesAtInstruction(
+      FindInstruction(InstructionOffset),
+      ppLiveVariables);
+}
+
+STDMETHODIMP dxil_debug_info::DxcPixDxilDebugInfo::IsVariableInRegister(
+    _In_ DWORD InstructionOffset,
+    _In_ const wchar_t *VariableName)
+{
+  return S_OK;
```

Comment:
Should this do anything else? If not, why does the method exist?

Reply:
There are variables that don't have a home location - then this method would return false. I still haven't found cases, though, so I'm leaving this placeholder here